software/src/applets/SwitchSeq.h

```diff
@@ -0,0 +1,275 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Uses the memory space in OC_patterns to sequence. Like a mini sequins.
+// 4 sequences are played in parallel. The note values may be configured with sequins or an upcoming sequence editor. CV can manipulate the sequences in a variety of ways.
+//
+// There are two channels controlled with a single clock (GATE 1). Gate 2 is a reset to makes the internal sequwncer's step equal to 0.
+// Each channel has a few different modes, controllable with the encoder.
+//
+// Modes:
+// OCT:  you are in this mode when the sequence arrow is solid. The encoder will rotate through sequences and it will play the chosen sequence, with CV controlling the octave (by 1 or 2 octaves).
+// PICK: In this mode CV will pick which sequence is playing. Sampled after each clock.
+// RAND: The CV is ignored, and instead a random sequence is chosen at random.
+// QUAN: The internal sequences are ignored and the CV will be quanitzed to a global scale. Right now that's c-dorian and only configurable in code.
+
+#define PICK_MODE -1
+#define RAND_MODE -2
+#define QUAN_MODE -3
+
+#define OCTAVE_1 (12 << 7)
+
+// 5.5 volts to play well with pressure points
+#define PP_MAX_INPUT_CV (HEMISPHERE_MAX_INPUT_CV * 11 / 12)
+#define OCTAVE_RANGE 2
+
+class SwitchSeq : public HemisphereApplet
+{
+public:
+    const char *applet_name()
+    {
+        return "SwitchSeq";
+    }
+
+    void Start()
+    {
+        quantizer.Init();
+        quantizer.Configure(OC::Scales::GetScale(scale), 0xffff);
+    }
+
+    void Controller()
+    {
+        // global clock. Do it before reset.
+        if (Clock(0))
+        {
+            StartADCLag(0);
+        }
+
+        // reset
+        if (Clock(1))
+        {
+            step = 0;
+        }
+
+        // only make changes on a clock'd signal.
+        if (EndOfADCLag(0))
+        {
+            step = (step + 1) % 16;
+
+            ForEachChannel(ch)
+            {
+                if (mode[ch] == RAND_MODE)
+                {
+                    cv[ch] = random(0, PP_MAX_INPUT_CV);
+                }
+                else
+                {
+                    cv[ch] = In(ch);
+                }
+                int32_t pitch = ValueForChannel(ch);
+                int32_t quantized = quantizer.Process(pitch, root << 7, 0);
```

Comment:
I have no idea why `root << 7` is here. It sounds good to me though.

Reply:
bitshift left by 7 is the same as multiplying by 128 - this converts semitones to full-range CV values.
I'm trying to standardize this in HemisphereApplet.h with functions like SetRootNote() and Quantize() instead of needing to manage a new braids::quantizer instance inside every applet...

Reply:
ahh yeah this makes sense. Thanks for the info. 

If we had global quantizer settings it would probably help with some of my comments - do you think that there's a place for that - similar to the shared clock - for each CV channel?